### PR TITLE
Added annotation @Step for methods in HeaderComponent class

### PR DESCRIPTION
**Pull request by kapustin-ilya**

### CODE REVIEW

**Review by IrenZh — APPROVED** [review]

**Review by lhalam — APPROVED** [review]

File: src/main/java/org/ssu/edu/teachua/ui/components/header/HeaderComponent.java [hunk 1]
```diff
@@ -109,23 +116,25 @@ public List<String> parseList() {
                 .map(WebElement::getText)
                 .collect(Collectors.toList());
     }
-
+    
+    @Step("Click needed location in drop-down menu.")
```
Comment (lhalam) [inline src/main/java/org/ssu/edu/teachua/ui/components/header/HeaderComponent.java:120]: add value to step `{location}`

File: src/main/java/org/ssu/edu/teachua/ui/components/header/HeaderComponent.java [hunk 2]
```diff
@@ -60,45 +60,52 @@ public class HeaderComponent extends BaseComponent {
...
 
-    @Step("click Challenges in header")
+    @Step("Click challenges in header. Open drop-down menu with all challenges.")
     public HeaderComponent clickChallengesButton() {
         waitForElementToBeClickable(challengesButton).click();
         return this;
     }
-
+    
+    @Step("Click needed challenge in drop-down menu. Open challenge page.")
```
Comment (lhalam) [inline src/main/java/org/ssu/edu/teachua/ui/components/header/HeaderComponent.java:82]: add value to step `{id}`

**Review by lhalam — COMMENTED** [review]
add value to steps

File: src/main/java/org/ssu/edu/teachua/ui/components/header/HeaderComponent.java (hunk 1 above)
Comment (kapustin-ilya) [reply src/main/java/org/ssu/edu/teachua/ui/components/header/HeaderComponent.java]: Done

File: src/main/java/org/ssu/edu/teachua/ui/components/header/HeaderComponent.java (hunk 2 above)
Comment (kapustin-ilya) [reply src/main/java/org/ssu/edu/teachua/ui/components/header/HeaderComponent.java]: Done

File: src/main/java/org/ssu/edu/teachua/ui/components/header/HeaderComponent.java [hunk 3]
```diff
@@ -137,3 +146,4 @@ public String getChallengeUrl(int id) {
         return challenge.getAttribute("href");
     }
```
Comment (TetianaFilatova) [inline src/main/java/org/ssu/edu/teachua/ui/components/header/HeaderComponent.java:147]: Add @Step with value {id}

Comment (kapustin-ilya) [reply src/main/java/org/ssu/edu/teachua/ui/components/header/HeaderComponent.java:148]: Done

**Review by TetianaFilatova — APPROVED** [review]

**Pull request merged**